### PR TITLE
fix querySelector method

### PR DESCRIPTION
**Pull request by YuichiNukiyama**
fixes [#12568](https://github.com/Microsoft/TypeScript/issues/12568)

### CODE REVIEW

**Comment by YuichiNukiyama** [comment]
done.

File: baselines/dom.generated.d.ts [hunk 1]
```diff
@@ -13736,8 +13736,10 @@ interface NavigatorUserMedia {
 
 interface NodeSelector {
     querySelector<K extends keyof ElementTagNameMap>(selectors: K): ElementTagNameMap[K] | null;
+    querySelector<E extends Element>(selectors: string): E | null;
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: why not combine both signatures as:
```ts
querySelector<E extends Element = Element>(selectors: string): E | null;
```

Comment (YuichiNukiyama) [inline baselines/dom.generated.d.ts]: I did not know generic default... I fix it 😃

**Comment by AustinMatherne** [comment]
@mhegazy Is there a holdup on getting this merged?

File: baselines/dom.generated.d.ts [hunk 2]
```diff
@@ -13736,8 +13736,9 @@ interface NavigatorUserMedia {
 
 interface NodeSelector {
     querySelector<K extends keyof ElementTagNameMap>(selectors: K): ElementTagNameMap[K] | null;
-    querySelector(selectors: string): Element | null;
+    querySelector<E extends Element = Element>(selectors: string): E | null;
     querySelectorAll<K extends keyof ElementListTagNameMap>(selectors: K): ElementListTagNameMap[K];
+    querySelectorAll<E extends Element>(selectors: string): NodeListOf<E>;
```
Comment (AustinMatherne) [inline baselines/dom.generated.d.ts]: Can we combine these, too?
```
querySelectorAll<E extends Element = Element>(selectors: string): NodeListOf<E>;
```

Comment (YuichiNukiyama) [inline baselines/dom.generated.d.ts]: Good point!

**Comment by AustinMatherne** [comment]
@mhegazy anything holding this PR up? This makes testing the DOM much nicer in Angular 2.

**Pull request merged**